keps/sig-network/2433-topology-aware-hints/README.md

```diff
@@ -196,29 +197,27 @@ running in.
 
 ### Configuration
 
-A new `service.kubernetes.io/topology-aware-routing` annotation will be
-supported on Services. This will have 2 potential values:
+The new Service `trafficPolicy` field will be expanded to support a new value:
 
-- `Auto`: When there are a sufficient number of endpoints for the Service, the
-  EndpointSlice controller will add topology hints for each endpoint that will
-  ensure a proportional amounts are available to each zone in a cluster.
-- `Disabled`: EndpointSlice hints will not be set for this feature.
+- `PreferZone`: When there are a sufficient number of endpoints for the Service,
+  the EndpointSlice controller will add topology hints for each endpoint that
+  will ensure a proportional amounts are available to each zone in a cluster.
 
-When this annotation is either unspecified or not set to `Auto`, the `Disabled`
-behavior will be default. A future KEP will explore changing the default
-behavior from `Disabled` to `Auto` with a new feature gate.
+A future KEP will explore changing the default value of this field to
+`PreferZone`.
 
 #### Interoperability
 
-If both this annotation and the deprecated `topologyKeys` field are set on a
-Service, `topologyKeys` will be given precedence and this annotation will be
-ignored. This will be true until the `topologyKeys` field is removed in the
-future.
+Validation will ensure that `trafficPolicy` can not be set to `PreferZone` when
+the deprecated `topologyKeys` field is also set. This will be true until the
+`topologyKeys` field is removed in the future.
 
 #### Feature Gate
 
-This functionality will be guarded by the `TopologyAwareHints` feature
-gate. This gate should not be enabled if the `ServiceTopology` gate is enabled.
+This functionality will be guarded by the `TopologyAwareHints` feature gate.
+This gate will be dependent on the `ServiceInternalTrafficPolicy` feature gate
+since it uses the `TrafficPolicy` guarded by that gate. This gate should not be
+enabled if the deprecated `ServiceTopology` gate is enabled.
```

Comment:
LOL - can we get a table or something?

Reply:
Haha, cleaned this up so it's hopefully a bit easier to understand.